From a70564d4fddb9138ff50846289b21a22672020de Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@nowhere.nt>
Date: Sat, 31 Jul 2021 18:50:07 +0200
Subject: [PATCH] error.c: Add missing stddef include.

---
 src/libutil/error.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libutil/error.c b/src/libutil/error.c
index 7212748..fcc0e3b 100644
--- a/src/libutil/error.c
+++ b/src/libutil/error.c
@@ -27,7 +27,7 @@
  * other dealings in this Software without prior written authorization from
  * Silicon Graphics, Inc.
  */
-
+#include <stddef.h>
 #include "gluos.h"
 #include "gluint.h"
 #include <GL/glu.h>
-- 
2.30.2

