From 2a9091d802bb30d88e349b4ca6ee32157f78ae14 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas@nowhere.net>
Date: Sat, 4 Aug 2018 07:51:10 +0200
Subject: [PATCH] Linux: Disable file access dialog on flatpak.

---
 src/chart1.cpp | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/src/chart1.cpp b/src/chart1.cpp
index f98038b4..1f250b23 100755
--- a/src/chart1.cpp
+++ b/src/chart1.cpp
@@ -11260,7 +11260,7 @@ bool CheckSerialAccess( void )
         }
     }
 
-#endif
+#endif  // 0
 
     //  Who owns /dev/ttyS0?
     bret = false;
@@ -11295,6 +11295,10 @@ You must install a serial port (modprobe correct kernel module) or plug in a usb
             bret = true;
     }
 
+#ifdef FLATPAK
+    return bret;
+#endif
+
     if(!bret){
 
         wxString msg = msg1 + _("\
@@ -11312,9 +11316,7 @@ You may do so by executing the following command from the linux command line:\n\
         OCPNMessageBox ( NULL, msg, wxString( _("OpenCPN Info") ), wxICON_INFORMATION | wxOK, 30 );
     }
 
-
-
-#endif
+#endif  // (__UNIX__) && !defined(__OCPN__ANDROID__)
 
     return bret;
 }
-- 
2.17.1

