diff --git a/mdns.c b//mdns.c
index 959453fd4..171854e44 100644
--- a/mdns.c
+++ b/mdns.c
@@ -31,6 +31,10 @@
 #undef recvfrom
 #endif

+#ifndef OCPN_ENABLE_MDNS_DEBUG
+#define printf(...)
+#endif
+
 static char addrbuffer[64];
 static char entrybuffer[256];
 static char namebuffer[256];
@@ -1159,6 +1163,8 @@ void signal_handler(int signal) {
 }
 #endif

+#if 0
+
 int
 main(int argc, const char* const* argv) {
 	int mode = 0;
@@ -1262,3 +1268,5 @@ main(int argc, const char* const* argv) {

 	return 0;
 }
+
+#endif  // 0
