From b924af96154f2ba755358ed0a7ad314396029bfa Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas@nowhere.net>
Date: Wed, 4 Jul 2018 10:59:09 +0200
Subject: [PATCH] install: Install plugin header file.

---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4a859bf9..479e8c24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2110,6 +2110,8 @@ ELSE(PACKAGE_FORMAT EQUAL "DEB")
 	INSTALL(FILES data/changelog DESTINATION ${PREFIX_DATA}/doc/${PACKAGE_NAME} PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
 ENDIF(PACKAGE_FORMAT EQUAL "DEB")
 
+INSTALL(FILES include/ocpn_plugin.h DESTINATION ${PREFIX_INCLUDE})
+
 INSTALL(DIRECTORY data/sounds/ DESTINATION ${PREFIX_PKGDATA}/sounds )
 
 IF(BUNDLE_DOCS MATCHES "ON")
-- 
2.17.0

