From: Alec Leamas <leamas.alec@gmail.com>
Date: Mon, 29 Oct 2018 10:24:38 -0400
Subject: Patch: Initialize display to x11 on wayland hosts (#1166).

Bug: https://github.com/OpenCPN/OpenCPN/issues/1166
---
 include/chart1.h |  1 +
 src/chart1.cpp   | 18 ++++++++++++++++++
 2 files changed, 19 insertions(+)

diff --git a/include/chart1.h b/include/chart1.h
index ad448da..192be01 100644
--- a/include/chart1.h
+++ b/include/chart1.h
@@ -310,6 +310,7 @@ private:
 class MyApp: public wxApp
 {
   public:
+    MyApp();
     bool OnInit();
     int OnExit();
     void OnInitCmdLine(wxCmdLineParser& parser);
diff --git a/src/chart1.cpp b/src/chart1.cpp
index d83e133..0b78cd7 100644
--- a/src/chart1.cpp
+++ b/src/chart1.cpp
@@ -32,6 +32,11 @@
 //#include "c:\\Program Files\\visual leak detector\\include\\vld.h"
 #endif
 
+// #1166 temporary fix.
+#ifdef __linux__
+#include <gdk/gdk.h>
+#include <gtk/gtk.h>
+#endif
 
 #include <wx/settings.h>
 #include <wx/print.h>
@@ -1552,6 +1557,19 @@ void ParseAllENC(wxWindow* parent)
         delete prog;
 }
 
+MyApp::MyApp()
+{
+#ifdef __linux__
+// Handle e. g., wayland default display -- see #1166.
+
+#if GTK_CHECK_VERSION(3, 10, 0)
+    gdk_set_allowed_backends("x11");
+#else
+    setenv("GDK_BACKEND", "x11", 1);
+#endif
+
+#endif   // __linux__
+}
 
 bool MyApp::OnInit()
 {
