From: Alec Leamas <leamas.alec@gmail.com>
Date: Mon, 29 Oct 2018 10:24:38 -0400
Subject: Patch: Initialize display to x11 on wayland hosts (#1166).

Bug: https://github.com/OpenCPN/OpenCPN/issues/1166
---
 include/chart1.h |  1 +
 src/chart1.cpp   | 18 ++++++++++++++++++
 2 files changed, 19 insertions(+)

Index: OpenCPN/include/chart1.h
===================================================================
--- OpenCPN.orig/include/chart1.h	2019-11-19 08:52:54.350637312 -0500
+++ OpenCPN/include/chart1.h	2019-11-19 08:52:54.346635273 -0500
@@ -310,6 +310,7 @@
 class MyApp: public wxApp
 {
   public:
+    MyApp();
     bool OnInit();
     int OnExit();
     void OnInitCmdLine(wxCmdLineParser& parser);
Index: OpenCPN/src/chart1.cpp
===================================================================
--- OpenCPN.orig/src/chart1.cpp	2019-11-19 08:52:54.350637312 -0500
+++ OpenCPN/src/chart1.cpp	2019-11-19 08:52:54.346635273 -0500
@@ -32,6 +32,11 @@
 //#include "c:\\Program Files\\visual leak detector\\include\\vld.h"
 #endif
 
+// #1166 temporary fix.
+#ifdef __linux__
+#include <gdk/gdk.h>
+#include <gtk/gtk.h>
+#endif
 
 #include <wx/settings.h>
 #include <wx/print.h>
@@ -1552,6 +1557,19 @@
         delete prog;
 }
 
+MyApp::MyApp()
+{
+#ifdef __linux__
+// Handle e. g., wayland default display -- see #1166.
+
+#if GTK_CHECK_VERSION(3, 10, 0)
+    gdk_set_allowed_backends("x11");
+#else
+    setenv("GDK_BACKEND", "x11", 1);
+#endif
+
+#endif   // __linux__
+}
 
 bool MyApp::OnInit()
 {
