From: Alec Leamas <leamas.alec@gmail.com>
Date: Mon, 27 Aug 2018 07:00:47 +0200
Subject: Build: Use external wxsvg library if available.

---
 CMakeLists.txt | 25 ++++++++++++++++++-------
 1 file changed, 18 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ba81181..e7fbf65 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -207,6 +207,7 @@ ELSE(UNIX)
   SET(USE_BUNDLED_LIBS "ON" CACHE BOOL "Use bundled libraries instead of system's")
 ENDIF(UNIX)
 
+
 IF(APPLE OR WIN32)
   IF(NOT BUNDLE_DOCS)
     SET (BUNDLE_DOCS "ON")
@@ -871,13 +872,23 @@ IF(OCPN_USE_SVG)
         )
         ADD_DEFINITIONS(-DwxsvgUSE_EXIF) 
     endif()
-    ADD_LIBRARY(WXSVG ${SRC_WXSVG})
-    TARGET_LINK_LIBRARIES(WXSVG ${SVG_LIBS})
-    INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/include/wxSVG)
-    INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/include/wxSVGXML)
-    INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/include)
-    INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/src)
-    SET(EXTRA_LIBS ${EXTRA_LIBS} WXSVG)
+
+    PKG_SEARCH_MODULE(WXSVG libwxsvg wxsvg)
+    IF (WXSVG_FOUND AND USE_BUNDLED_LIBS MATCHES "ON")
+      message (STATUS "Building with system wxsvg includes")
+      INCLUDE_DIRECTORIES(${WXSVG_INCLUDE_DIR})
+      SET(EXTRA_LIBS ${EXTRA_LIBS} ${WXSVG_LIBRARIES})
+    ELSE (WXSVG_FOUND AND USE_BUNDLED_LIBS MATCHES "ON")
+      message (STATUS "Building with bundled wxsvg includes")
+      ADD_LIBRARY(WXSVG ${SRC_WXSVG})
+      TARGET_LINK_LIBRARIES(WXSVG ${SVG_LIBS})
+      INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/include/wxSVG)
+      INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/include/wxSVGXML)
+      INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/include)
+      INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/wxsvg/src)
+      set_property(TARGET WXSVG PROPERTY COMPILE_FLAGS "${OBJ_VISIBILITY}")
+      SET(EXTRA_LIBS ${EXTRA_LIBS} WXSVG)
+    ENDIF (WXSVG_FOUND AND USE_BUNDLED_LIBS MATCHES "ON")
   ENDIF(NOT QT_ANDROID)
 ENDIF(OCPN_USE_SVG)
 
