From: Alec Leamas <leamas.alec@gmail.com>
Date: Sun, 19 Aug 2018 09:34:31 +0200
Subject: Build: Use system lz4 libs if available.

Origin: upstream, https://github.com/OpenCPN/OpenCPN/commit/9c8c28dbfc4d
Applied-Upstream: 5.0.0
---
 CMakeLists.txt | 21 +++++++++++++++------
 1 file changed, 15 insertions(+), 6 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8a30753..7b990ab 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1499,10 +1499,7 @@ SET(EXTRA_LIBS ${EXTRA_LIBS} SYMBOLS)
 
 # Compile texcompression support library
 IF (OPENGL_FOUND)
-
   SET(SRC_TEXCMP
-    src/texcmp/lz4/lz4.c
-    src/texcmp/lz4/lz4hc.c
     src/texcmp/squish/alpha.cpp
     src/texcmp/squish/clusterfit.cpp
     src/texcmp/squish/colourblock.cpp
@@ -1514,10 +1511,22 @@ IF (OPENGL_FOUND)
     src/texcmp/squish/singlecolourfitfast.cpp
     src/texcmp/squish/twocolourfitfast.cpp
     src/texcmp/squish/squish.cpp
-    src/texcmp/etcpak.cpp)
-
-  INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/texcmp/lz4)
+    src/texcmp/etcpak.cpp
+  )
   INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/texcmp/squish)
+  PKG_SEARCH_MODULE(LZ4 liblz4 lz4)
+  IF (LZ4_FOUND AND USE_BUNDLED_LIBS MATCHES "OFF")
+    message (STATUS "Building with system lz4")
+    INCLUDE_DIRECTORIES(${LZ4_INCLUDE_DIR})
+    SET(EXTRA_LIBS ${EXTRA_LIBS} ${LZ4_LIBRARIES})
+  ELSE (LZ4_FOUND AND USE_BUNDLED_LIBS MATCHES "OFF")
+    message (STATUS "Building with bundled lz4")
+    SET(SRC_TEXCMP ${SRC_TEXCMP}
+      src/texcmp/lz4/lz4.c
+      src/texcmp/lz4/lz4hc.c
+    )
+    INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/src/texcmp/lz4)
+  ENDIF (LZ4_FOUND AND USE_BUNDLED_LIBS MATCHES "OFF")
 
   ADD_LIBRARY(TEXCMP ${SRC_TEXCMP})
   SET(EXTRA_LIBS ${EXTRA_LIBS} TEXCMP)
