From bdb252a831be842ea02d27fe5ab865feee255eb8 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 10 Aug 2024 09:49:04 +0200
Subject: [PATCH 4/6] tests: disable rest server tests for now

Bug: https://github.com/OpenCPN/OpenCPN/issues/4090
---
 test/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 6b70259ee..cf1c76f45 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -32,7 +32,7 @@ if (MSVC)
   set(CURL_HINTS C:\\mingw64\\bin C:\\ProgramData\\chocolatey\\bin)
 endif ()
 find_program(CURL NAMES curl HINTS ${CURL_HINTS})
-if (CURL)
+if (CURL AND NOT OCPN_DISTRO_BUILD)
   list(APPEND SRC rest-tests.cpp)
 endif ()
 
-- 
2.34.1

