From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 2 May 2020 10:39:17 +0200
Subject: build: gtk:  Ensure linking to required gtk libs.

Bug: https://github.com/OpenCPN/OpenCPN/issues/1889
---
 CMakeLists.txt | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index dbad89d..7caac09 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -611,9 +611,9 @@ endif (OCPN_BUNDLE_DOCS MATCHES "ON")
 #
 # Linux: set up GTK (wxWidgets prerequisite).
 #
-#if (NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
-#  include(OcpnFindGtk)
-#endif ()
+if (NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
+  include(OcpnFindGtk)
+endif ()
 
 # set a build flag for arm architecture, to catch any rPI runtime changes
 # required
@@ -1181,8 +1181,8 @@ target_link_libraries(${PACKAGE_NAME} PRIVATE ssl::sha1)
 # Linux: set up GTK
 #
 if (NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
-#  include(OcpnFindGtk)
-#  target_link_libraries(${PACKAGE_NAME} PRIVATE ocpn::gtk)
+  include(OcpnFindGtk)
+  target_link_libraries(${PACKAGE_NAME} PRIVATE ocpn::gtk)
 endif (NOT WIN32 AND NOT APPLE AND NOT QT_ANDROID)
 
 # Search for opengles, short of running a program to test the speed of
