From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 1 Jan 2022 01:08:33 +0100
Subject: CMakeLists: Add effective target debian -> debian-gtk3

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b115328..aa8be50 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1819,7 +1819,7 @@ endif (UNIX AND NOT APPLE)
 # Finalize config.h setup and deploy it.
 include(TargetSetup)
 if ("${wxWidgets_LIBRARIES}" MATCHES "gtk3u")
-  string(REPLACE ubuntu ubuntu-gtk3 PKG_TARGET ${PKG_TARGET})
+  string(REPLACE debian debian-gtk3 PKG_TARGET ${PKG_TARGET})
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D__WXGTK3__")
 endif ()
 configure_file(${CMAKE_SOURCE_DIR}/cmake/in-files/config.h.in ${CMAKE_BINARY_DIR}/include/config.h)
