From: Alec Leamas <leamas.alec@gmail.com>
Date: Wed, 20 Nov 2019 15:47:38 -0500
Subject: CMakelists: move appdata.xml to /usr/share/metainfo

Bug: https://github.com/OpenCPN/OpenCPN/issues/1490
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ed290ac..06198ef 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2421,7 +2421,7 @@ IF(NOT APPLE)
     INSTALL(FILES src/bitmaps/opencpn-64.png DESTINATION ${PREFIX_DATA}/icons/hicolor/64x64/apps RENAME opencpn.png PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
     INSTALL(FILES src/bitmaps/other_svg_src/opencpn.svg DESTINATION ${PREFIX_DATA}/icons/hicolor/scalable/apps PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ)
     INSTALL(FILES data/opencpn.desktop DESTINATION ${PREFIX_DATA}/applications PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
-    INSTALL(FILES data/opencpn.appdata.xml DESTINATION ${PREFIX_DATA}/appdata PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
+    INSTALL(FILES data/opencpn.appdata.xml DESTINATION ${PREFIX_DATA}/metainfo PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
     INSTALL(FILES opencpn.1 DESTINATION ${PREFIX_DATA}/man/man1 PERMISSIONS OWNER_READ OWNER_WRITE GROUP_READ WORLD_READ )
   ENDIF(UNIX)
 
