From: Alec Leamas <leamas.alec@gmail.com>
Date: Sat, 1 Jan 2022 01:22:17 +0100
Subject: PluginHandler: Accept ubuntu-gtk3/18.04 plugins.

---
 src/PluginHandler.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/PluginHandler.cpp b/src/PluginHandler.cpp
index 913d8de..eaefe74 100644
--- a/src/PluginHandler.cpp
+++ b/src/PluginHandler.cpp
@@ -199,6 +199,12 @@ public:
         "10;ubuntu-x86_64;18.04",
         "11;ubuntu-gtk3-x86_64;20.04",
         "sid;ubuntu-gtk3-x86_64;20.04"};
+    if (m_abi == "debian-gtk3") {
+      // Corner case on debian 10 only:
+      if (plugin.abi_version() == "18.04" && plugin.abi() == "ubuntu-gtk3") {
+        return true;
+      }
+    }
     if (ocpn::startswith(m_abi, "debian-x86_64")) {
       wxLogDebug("Checking for debian and ubuntu, debian-x86_64 host");
       const std::string host_version =
