From: Alec Leamas <leamas.alec@gmail.com>
Date: Wed, 28 Dec 2022 07:26:42 +0100
Subject: libs/gdal/.../cpl_config.h Don't undef unconfigured stuff (#2896)

Bug: https://github.com/OpenCPN/OpenCPN/issues/2896
Forwarded: https://github.com/OpenCPN/OpenCPN/pull/2897
---
 libs/gdal/include/gdal/cpl_config.h | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/libs/gdal/include/gdal/cpl_config.h b/libs/gdal/include/gdal/cpl_config.h
index 7932123..25c5150 100644
--- a/libs/gdal/include/gdal/cpl_config.h
+++ b/libs/gdal/include/gdal/cpl_config.h
@@ -53,29 +53,9 @@
 #  define STDC_HEADERS
 #endif
 
-/* Define if you have the <fcntl.h> header file.  */
-#undef HAVE_FCNTL_H
-
-/* Define if you have the <unistd.h> header file.  */
-#undef HAVE_UNISTD_H
-
-/* Define if you have the <stdint.h> header file.  */
-#undef HAVE_STDINT_H
-
-#undef HAVE_LIBDL
-
 /* Define to 1 if you have the <locale.h> header file. */
 #define HAVE_LOCALE_H 1
 
-/* Define to 1 if you have the `localtime_r' function. */
-#undef HAVE_LOCALTIME_R
-
-#undef HAVE_DLFCN_H
-#undef HAVE_DBMALLOC_H
-#undef HAVE_LIBDBMALLOC
-#undef WORDS_BIGENDIAN
-
 #define CPL_DISABLE_DLL
 
-
 #endif
