From aac94163bb4ec1b51d83127cb0d35eea71cd26ea Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@nowhere.net>
Date: Tue, 27 Dec 2022 13:13:34 +0100
Subject: [PATCH] CMakeLists, tests: Don't use serial library if not configured

---
 CMakeLists.txt      | 4 +++-
 test/CMakeLists.txt | 4 +++-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 54accb698..d64fb4271 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3047,10 +3047,12 @@ if (NOT QT_ANDROID AND NOT APPLE)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::nmea0183)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::pugixml)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::s52plib)
-  target_link_libraries(opencpn-cmd PRIVATE ocpn::serial)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::sound)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::tinyxml)
   target_link_libraries(opencpn-cmd PRIVATE ocpn::wxjson)
+  if (OCPN_USE_NEWSERIAL)
+    target_link_libraries(opencpn-cmd PRIVATE ocpn::serial)
+  endif ()
 
   target_include_directories(opencpn-cmd PUBLIC "${LIBELF_INCLUDE_DIR}")
   target_link_libraries(opencpn-cmd PRIVATE "${LIBELF_LIBRARY}")
diff --git a/test/CMakeLists.txt b/test/CMakeLists.txt
index 3701119d0..c70e63907 100644
--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -57,7 +57,6 @@ target_link_libraries(tests PRIVATE ocpn::N2KParser)
 target_link_libraries(tests PRIVATE ocpn::iso8211)
 #target_link_libraries(tests PRIVATE ocpn::wxsvg)
 target_link_libraries(tests PRIVATE ocpn::garminhost)
-target_link_libraries(tests PRIVATE ocpn::serial)
 target_link_libraries(tests PRIVATE ocpn::easywsclient)
 target_link_libraries(tests PRIVATE ocpn::tinyxml)
 target_link_libraries(tests PRIVATE observable::observable)
@@ -65,6 +64,9 @@ target_link_libraries(tests PRIVATE ocpn::mongoose)
 target_link_libraries(tests PRIVATE ocpn::geoprim)
 target_link_libraries(tests PRIVATE ocpn::s52plib)
 target_link_libraries(tests PRIVATE ocpn::pugixml)
+if (OCPN_USE_NEWSERIAL)
+  target_link_libraries(tests PRIVATE ocpn::serial)
+endif ()
 
 if (DEFINED CURL_LIBRARIES)
   target_link_libraries(tests PRIVATE ${CURL_LIBRARIES})
-- 
2.38.1

