From a3219d14d3251baa9c25e6e9a582b3916ed49df6 Mon Sep 17 00:00:00 2001
From: Alec Leamas <leamas.alec@gmail.com>
Date: Fri, 23 May 2025 14:15:58 +0200
Subject: [PATCH 6/6] libdnet: Handle cmake < 3.20

---
 libs/libdnet/cmake/PatchFile.cmake | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libs/libdnet/cmake/PatchFile.cmake b/libs/libdnet/cmake/PatchFile.cmake
index 70dfec863..f792f1bb0 100644
--- a/libs/libdnet/cmake/PatchFile.cmake
+++ b/libs/libdnet/cmake/PatchFile.cmake
@@ -29,7 +29,11 @@ find_package(Git REQUIRED)
 
 set(CMAKE_EXECUTE_PROCESS_COMMAND_ECHO STDOUT)
 
-cmake_path(GET patch FILENAME patch_name)
+if (${CMAKE_VERSION} VERSION_LESS 3.20.0)
+  get_filename_component(patch_name ${patch} NAME)
+else ()
+  cmake_path(GET patch FILENAME patch_name)
+endif ()
 
 execute_process(
   COMMAND ${GIT_EXECUTABLE} --git-dir= apply -p1 --ignore-whitespace "${patch}"
-- 
2.34.1

