From: Alec Leamas <leamas.alec@gmail.com>
Date: Sun, 3 May 2020 10:36:57 +0200
Subject: wxserverdisc: Disable pointer-sign warnings.

Bug: https://github.com/OpenCPN/OpenCPN/issues/1267
---
 libs/wxservdisc/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libs/wxservdisc/CMakeLists.txt b/libs/wxservdisc/CMakeLists.txt
index 7163eac..c2c3991 100644
--- a/libs/wxservdisc/CMakeLists.txt
+++ b/libs/wxservdisc/CMakeLists.txt
@@ -15,6 +15,9 @@ add_library(ocpn::wxservdisc ALIAS WXSERVDISC)
 
 set_property(TARGET WXSERVDISC PROPERTY COMPILE_FLAGS "${OBJ_VISIBILITY}")
 target_include_directories(WXSERVDISC PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  target_compile_options(WXSERVDISC PRIVATE -Wno-pointer-sign -Wno-parentheses)
+endif ()
 
 if (WIN32)
   target_link_libraries(WXSERVDISC PRIVATE ws2_32)
