From: Alec Leamas <leamas.alec@nowhere.net>
Date: Fri, 2 Aug 2024 22:11:47 +0200
Subject: tests: disable test with intermittent failures

---
 test/tests.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/test/tests.cpp b/test/tests.cpp
index 697a098..c9f0488 100644
--- a/test/tests.cpp
+++ b/test/tests.cpp
@@ -765,7 +765,9 @@ public:
 };
 #endif  // __unix__
 
+#ifndef OCPN_DISTRO_BUILD
 TEST(MsgFixture, ObservableMsg) { MsgCliApp app; };
+#endif
 
 TEST(Messaging, NavMsg) { TransportCliApp app; };
 
