From: Alec Leamas <leamas.alec@gmail.com>
Date: Mon, 15 Feb 2021 12:46:13 +0100
Subject: testing using glx.h

---
 src/glChartCanvas.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/glChartCanvas.cpp b/src/glChartCanvas.cpp
index 059d989..45cc856 100644
--- a/src/glChartCanvas.cpp
+++ b/src/glChartCanvas.cpp
@@ -61,6 +61,8 @@ private:
 #include <GL/glx.h>
 #endif
 
+#include <GL/glx.h>
+
 #include "dychart.h"
 
 #include "glChartCanvas.h"
