From: Alec Leamas <leamas.alec@gmail.com>
Date: Mon, 25 Aug 2025 10:38:24 +0200
Subject: OCPNRegion: Fix uninitialized variable FTBFS

---
 gui/src/OCPNRegion.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gui/src/OCPNRegion.cpp b/gui/src/OCPNRegion.cpp
index bb57b65..487f644 100644
--- a/gui/src/OCPNRegion.cpp
+++ b/gui/src/OCPNRegion.cpp
@@ -2861,7 +2861,7 @@ OGdkRegion *gdk_region_polygon(const OGdkPoint *points, int n_points,
   OEdgeTable ET = {0};         /* header node for ET      */
   OEdgeTableEntry AET;         /* header node for AET     */
   OEdgeTableEntry *pETEs;      /* EdgeTableEntries pool   */
-  OScanLineListBlock SLLBlock; /* header for scanlinelist */
+  OScanLineListBlock SLLBlock = {0}; /* header for scanlinelist */
   int fixWAET = FALSE;
   OPOINTBLOCK FirstPtBlock, *curPtBlock; /* PtBlock buffers    */
   OPOINTBLOCK *tmpPtBlock;
