From: Alec Leamas <leamas.alec@gmail.com>
Date: Fri, 23 May 2025 12:02:47 +0200
Subject: tests: disable tests requiring locale

---
 test/datetime_tests.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/test/datetime_tests.cpp b/test/datetime_tests.cpp
index d6da1f7..040f112 100644
--- a/test/datetime_tests.cpp
+++ b/test/datetime_tests.cpp
@@ -212,7 +212,7 @@ TEST_F(DateTimeFormatTest, ShowTimezoneDefault) {
   EXPECT_EQ(result, "2023-01-22 12:45:57 UTC");
 }
 
-#ifndef DISABLE_LOCALE_TESTS
+#ifndef OCPN_DISTRO_BUILD
 // Test with Local Time in EST timezone
 TEST_F(DateTimeFormatTest, LocalTimezoneEST) {
   // Set timezone to EST for this test (UTC-5)
