--- a/include/ais.h
+++ b/include/ais.h
@@ -110,7 +110,6 @@
  *
  */
 
-
 #include "wx/wxprec.h"
 
 #ifndef  WX_PRECOMP
@@ -118,7 +117,18 @@
 #endif //precompiled headers
 
 #include <wx/datetime.h>
+
+////////////TH//////////////////
+#ifdef __WXGTK__
+#define GSocket GlibGSocket
+#include <gtk/gtk.h>
+//#include <glib.h>
 #include "wx/socket.h"
+#undef GSocket
+#else
+#include "wx/socket.h"
+#endif
+///////////TH100126////////////////
 #include "wx/sound.h"
 
 #include "dychart.h"
--- a/include/nmea.h
+++ b/include/nmea.h
@@ -1,5 +1,5 @@
 /******************************************************************************
- * $Id: nmea.h,v 1.24 2010/01/02 01:58:12 bdbcat Exp $
+ * $Id: nmea.h,v 1.25 2010/01/25 01:31:53 badfeed Exp $
  *
  * Project:  OpenCP
  * Purpose:  NMEA Data Object
@@ -26,6 +26,9 @@
  ***************************************************************************
  *
  * $Log: nmea.h,v $
+ * Revision 1.25  2010/01/25 01:31:53  badfeed
+ * Re-enable GSocket rename workaround for gtk
+ *
  * Revision 1.24  2010/01/02 01:58:12  bdbcat
  * Clone Event
  *
@@ -92,7 +95,6 @@
 #ifndef __NMEA_H__
 #define __NMEA_H__
 
-// Include wxWindows' headers
 
 #include "wx/wxprec.h"
 
@@ -101,19 +103,20 @@
 #endif //precompiled header
 
 //#include <gio/gio.h>
+// Include wxWindows' headers
+
+/////////////TH////////////////////////
 
 #ifdef __WXGTK__
-// newer versions of glib define its own GSocket but we unfortunately use this
-// name in our own (semi-)public header and so can't change it -- rename glib
-// one instead
-//#define GSocket GlibGSocket
-//#include <gdk/gdk.h>
+#define GSocket GlibGSocket
+#include <gtk/gtk.h>
 //#include <glib.h>
-//#undef GSocket
-#endif
-
-
 #include "wx/socket.h"
+#undef GSocket
+#else
+#include "wx/socket.h"
+#endif
+////////////////////TH100126/////////////////
 #include <wx/datetime.h>
 
 
