--- a/Makefile.in
+++ b/Makefile.in
@@ -2449,30 +2449,30 @@
 install-data-local:
 	mkdir -p $(DESTDIR)$(pkgdatadir)
 	mkdir -p $(DESTDIR)$(pkgdatadir)/wvsdata
-	cp -p ./$(srcdir)/data/wvsdata/wvs1.dat $(DESTDIR)$(pkgdatadir)/wvsdata
-	cp -p ./$(srcdir)/data/wvsdata/wvs43.dat $(DESTDIR)$(pkgdatadir)/wvsdata
-	cp -p ./$(srcdir)/data/wvsdata/readmewvs.txt $(DESTDIR)$(pkgdatadir)/wvsdata
+	install -p -m 644 ./$(srcdir)/data/wvsdata/wvs1.dat $(DESTDIR)$(pkgdatadir)/wvsdata
+	install -p -m 644 ./$(srcdir)/data/wvsdata/wvs43.dat $(DESTDIR)$(pkgdatadir)/wvsdata
+	install -p -m 644 ./$(srcdir)/data/wvsdata/readmewvs.txt $(DESTDIR)$(pkgdatadir)/wvsdata
 	mkdir -p $(DESTDIR)$(pkgdatadir)/s57data
-	cp -p ./$(srcdir)/data/s57data/s57attributes.csv $(DESTDIR)$(pkgdatadir)/s57data
-	cp -p ./$(srcdir)/data/s57data/s57objectclasses.csv $(DESTDIR)$(pkgdatadir)/s57data
-	cp -p ./$(srcdir)/data/s57data/s57expectedinput.csv $(DESTDIR)$(pkgdatadir)/s57data
-	cp -p ./$(srcdir)/data/s57data/attdecode.csv $(DESTDIR)$(pkgdatadir)/s57data
-	cp -p ./$(srcdir)/data/s57data/S52RAZDS.RLE $(DESTDIR)$(pkgdatadir)/s57data
+	install -p -m 644 ./$(srcdir)/data/s57data/s57attributes.csv $(DESTDIR)$(pkgdatadir)/s57data
+	install -p -m 644 ./$(srcdir)/data/s57data/s57objectclasses.csv $(DESTDIR)$(pkgdatadir)/s57data
+	install -p -m 644 ./$(srcdir)/data/s57data/s57expectedinput.csv $(DESTDIR)$(pkgdatadir)/s57data
+	install -p -m 644 ./$(srcdir)/data/s57data/attdecode.csv $(DESTDIR)$(pkgdatadir)/s57data
+	install -p -m 644 ./$(srcdir)/data/s57data/S52RAZDS.RLE $(DESTDIR)$(pkgdatadir)/s57data
 	mkdir -p $(DESTDIR)$(pkgdatadir)/tcdata
-	cp -p ./$(srcdir)/data/tcdata/HARMONIC  $(DESTDIR)$(pkgdatadir)/tcdata
-	cp -p ./$(srcdir)/data/tcdata/HARMONIC.IDX  $(DESTDIR)$(pkgdatadir)/tcdata
-	cp -p ./$(srcdir)/data/license.txt  $(DESTDIR)$(pkgdatadir)
+	install -p -m 644 ./$(srcdir)/data/tcdata/HARMONIC  $(DESTDIR)$(pkgdatadir)/tcdata
+	install -p -m 644 ./$(srcdir)/data/tcdata/HARMONIC.IDX  $(DESTDIR)$(pkgdatadir)/tcdata
+	install -p -m 644 ./$(srcdir)/data/license.txt  $(DESTDIR)$(pkgdatadir)
 	mkdir -p $(DESTDIR)$(pkgdatadir)/doc
-	cp -p ./$(srcdir)/data/doc/help.html  $(DESTDIR)$(pkgdatadir)/doc
+	install -p -m 644 ./$(srcdir)/data/doc/help.html  $(DESTDIR)$(pkgdatadir)/doc
 	mkdir -p $(DESTDIR)$(pkgdatadir)/doc/images
-	cp -p ./$(srcdir)/data/doc/images/*.jpg  $(DESTDIR)$(pkgdatadir)/doc/images
+	install -p -m 644 ./$(srcdir)/data/doc/images/*.jpg  $(DESTDIR)$(pkgdatadir)/doc/images
 	mkdir -p $(DESTDIR)$(pkgdatadir)/doc/includes
-	cp -p ./$(srcdir)/data/doc/includes/*.css  $(DESTDIR)$(pkgdatadir)/doc/includes
-	cp -p ./$(srcdir)/data/doc/includes/*.js  $(DESTDIR)$(pkgdatadir)/doc/includes
+	install -p -m 644 ./$(srcdir)/data/doc/includes/*.css  $(DESTDIR)$(pkgdatadir)/doc/includes
+	install -p -m 644 ./$(srcdir)/data/doc/includes/*.js  $(DESTDIR)$(pkgdatadir)/doc/includes
 	mkdir -p $(DESTDIR)$(datadir)/icons
-	cp -p ./$(srcdir)/data/opencpn.png  $(DESTDIR)$(datadir)/icons
+	install -p -m 644 ./$(srcdir)/data/opencpn.png  $(DESTDIR)$(datadir)/icons
 	mkdir -p $(DESTDIR)$(datadir)/applications
-	cp -p ./$(srcdir)/data/opencpn.desktop  $(DESTDIR)$(datadir)/applications
+	install -p -m 644 ./$(srcdir)/data/opencpn.desktop  $(DESTDIR)$(datadir)/applications
 	mkdir -p $(DESTDIR)$(datadir)/locale/de/LC_MESSAGES
 	mkdir -p $(DESTDIR)$(datadir)/locale/da/LC_MESSAGES
 	mkdir -p $(DESTDIR)$(datadir)/locale/it/LC_MESSAGES
@@ -2481,14 +2481,14 @@
 	mkdir -p $(DESTDIR)$(datadir)/locale/es/LC_MESSAGES
 	mkdir -p $(DESTDIR)$(datadir)/locale/nl/LC_MESSAGES
 	mkdir -p $(DESTDIR)$(datadir)/locale/sv/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_de_DE.mo $(DESTDIR)$(datadir)/locale/de/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_da_DK.mo $(DESTDIR)$(datadir)/locale/da/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_it_IT.mo $(DESTDIR)$(datadir)/locale/it/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_pl_PL.mo $(DESTDIR)$(datadir)/locale/pl/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_fr_FR.mo $(DESTDIR)$(datadir)/locale/fr/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_es_ES.mo $(DESTDIR)$(datadir)/locale/es/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_nl_NL.mo $(DESTDIR)$(datadir)/locale/nl/LC_MESSAGES
-	cp -p ./$(srcdir)/po/opencpn_sv_SE.mo $(DESTDIR)$(datadir)/locale/sv/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_de_DE.mo $(DESTDIR)$(datadir)/locale/de/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_da_DK.mo $(DESTDIR)$(datadir)/locale/da/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_it_IT.mo $(DESTDIR)$(datadir)/locale/it/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_pl_PL.mo $(DESTDIR)$(datadir)/locale/pl/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_fr_FR.mo $(DESTDIR)$(datadir)/locale/fr/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_es_ES.mo $(DESTDIR)$(datadir)/locale/es/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_nl_NL.mo $(DESTDIR)$(datadir)/locale/nl/LC_MESSAGES
+	install -p -m 644 ./$(srcdir)/po/opencpn_sv_SE.mo $(DESTDIR)$(datadir)/locale/sv/LC_MESSAGES
 
 uninstall-local:
 	rm -fr $(DESTDIR)$(pkgdatadir)/wvsdata/*
