Description: point to Debian-package doc dir
Author: Hamish B <hamish_b yahoo com>
Index: opencpn/src/about.cpp
===================================================================
--- opencpn.orig/src/about.cpp	2011-05-24 20:03:38.014938255 +1200
+++ opencpn/src/about.cpp	2011-05-24 20:03:45.214915189 +1200
@@ -449,7 +449,8 @@
             wxString def_lang_canonical = wxLocale::GetLanguageInfo(wxLANGUAGE_DEFAULT)->CanonicalName;
 
             wxString help_locn = _T("doc/help_");
-            help_locn.Prepend(*m_pDataLocn);
+//          help_locn.Prepend(*m_pDataLocn);
+	    help_locn.Prepend(_T("/usr/share/doc/opencpn-doc/"));
 
             wxString help_try = help_locn;
             help_try += def_lang_canonical;
