Description: point to Debian-package doc dir
Author: Hamish B <hamish_b yahoo com>
Index: opencpn/src/about.cpp
===================================================================
--- opencpn.orig/src/about.cpp	2011-08-02 22:59:00.202915243 +1200
+++ opencpn/src/about.cpp	2011-08-02 22:59:00.210927964 +1200
@@ -452,7 +452,8 @@
             wxString def_lang_canonical = wxLocale::GetLanguageInfo(wxLANGUAGE_DEFAULT)->CanonicalName;
 
             wxString help_locn = _T("doc/help_");
-            help_locn.Prepend(*m_pDataLocn);
+//          help_locn.Prepend(*m_pDataLocn);
+	    help_locn.Prepend(_T("/usr/share/doc/opencpn-doc/"));
 
             wxString help_try = help_locn;
             help_try += def_lang_canonical;
