Description: Use system's copy of GPL2.txt
Author: Hamish B <hamish_b yahoo com>
Index: opencpn/src/about.cpp
===================================================================
--- opencpn.orig/src/about.cpp	2011-04-25 09:18:19.000000000 +1200
+++ opencpn/src/about.cpp	2011-04-25 09:37:24.193610483 +1200
@@ -295,8 +295,9 @@
 
       pLicenseTextCtl->SetDefaultStyle(ta);
       pLicenseTextCtl->Clear();
-      wxString license_loc(*m_pDataLocn);
-      license_loc.Append(_T("license.txt"));
+      //wxString license_loc(*m_pDataLocn);
+      //license_loc.Append(_T("license.txt"));
+      wxString license_loc(_T("/usr/share/common-licenses/GPL-2"));
 
       wxTextFile license_file(license_loc);
 
