Description: Use system's copy of the TinyXML library
Author: Hamish B <hamish_b yahoo com>
Index: opencpn/include/gpxdocument.h
===================================================================
--- opencpn.orig/include/gpxdocument.h	2011-08-02 23:09:19.967165256 +1200
+++ opencpn/include/gpxdocument.h	2011-08-02 23:09:48.359953077 +1200
@@ -29,7 +29,9 @@
 
 #include <wx/wx.h>
 #include <tinyxml.h>
+#ifndef TIXML_USE_STL
 #include <tinystr.h>
+#endif
 #include <stdio.h>
 
 const wxString GpxxColorNames[] = { _("Black"), _("DarkRed"), _("DarkGreen"), _("DarkYellow"), _("DarkBlue"), _("DarkMagenta"), _("DarkCyan"), _("LightGray"), _("DarkGray"), _("Red"), _("Green"), _("Yellow"), _("Blue"), _("Magenta"), _("Cyan"), _("White") };//The last color defined by Garmin is transparent - we ignore it
